src/aiai_eval/leaderboard_utils.py

```diff
@@ -44,7 +44,16 @@ def get_task(self, task_name: str, raw: bool = False) -> dict:
             endpoint = f"{self.base_url}/{task_config.name}"
 
         # Get task from Leaderboard
-        task = self.get(endpoint).json()
+        response = self.get(endpoint)
+
+        # Check if we got a valid response and raise error if not
+        if response.status_code == 204 or not response.headers[
+            "content-type"
+        ].strip().startswith("application/json"):
+            raise ValueError(response.text)
```

Comment:
Maybe check for status code not equal to 200 here, to catch all potential errors?